tools/cr/commit-msg.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/env python3
+# Copyright (c) 2025 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at https://mozilla.org/MPL/2.0/.
+
+# This script has several convenience features when committing changes to
+# brave-core.
+#
+# 1. Chromium upgrade branches automatically added as tags
+#
+#    Any branch that starts with "cr" and is followed by a number will be added
+#    to the commit message as a tag (e.g. [cr100] Some change).
+#
+# 2. Tags can be passed in via the environment variable TAGS
+#
+#    For instance `TAGS=CodeHealth,canary git commit -m "Some change"` will
+#    add [CodeHealth] and [canary] to the commit message.
+#
+# 3. Labels in the branch name behind '+' will be added as tags
+#
+#    This means that a branch `canary+fix-failure` will have [canary] added to
+#    the commit message.
+#
+# 4. CULPRIT environment variable for google source links
+#
+#    If the CULPRIT environment variable is set, it will be used to add links to
+#    the commit message. The CULPRIT variable should be a comma-separated list
+#    of commit hashes. The script will look up the commit messages for each hash
+#    and add them to the commit message. The links will be in the format
+#    https://chromium.googlesource.com/chromium/src/+/{hash}.
+#
+# This script can be set up either as a commit-msg hook, or as a
+# prepare-commit-msg hook. The difference is that the commit-msg hook will run
+# after the commit message is created, while the prepare-commit-msg hook will
+# run before the commit message is created, which allows you to inspect the
+# insertions.
+#
+# To install this script as a commit-msg hook, run the following command:
+#   cp tools/cr/commit-msg.py .git/hooks/commit-msg
+#
+
+import os
+import sys
+import subprocess
+import re
+
+PREFIXES_FOR_UPGRADE_COMMITS = [
+    'Update from Chromium ',
+    'Conflict-resolved patches from Chromium ',
+    'Update patches from Chromium ',
+    'Updated strings for Chromium ',
+]
+
+
+def main():
+    if os.getenv("GIT_REBASE"):
```

Comment:
this has to be manually set or it's something git adds during a rebase?

Reply:
Actually this check is wrong. I'm now checking for the files git create when it is doing a rebase or cherry-pick. Fixed.